src/megaapi_impl.cpp

```diff
@@ -2916,7 +2916,27 @@ MegaNode *MegaNodeListPrivate::get(int i)
 
 int MegaNodeListPrivate::size()
 {
-	return s;
+    return s;
+}
+
+void MegaNodeListPrivate::addNode(MegaNode *node)
+{
+    MegaNodeList *duplicatedList = copy();
+
+    s = 0;
+    for(int i=0; i<s; i++)
+        delete list[i];
```

Comment:
This is very dangerous. According to the documentation, internal MegaNode objects are supposed to be valid until the deletion of the MegaNodeList and this breaks that behavior. It should be avoided (for example by creating a new list of pointers but reusing the pointers from the previous list) or, if there isn't any other possible solution, the documentation should be updated to say that the usage of this function invalidate all internal MegaNode objects.